contracts/RewardManager.sol

```diff
@@ -319,13 +345,25 @@ contract RewardManager is Ownable, Versionable, Safe {
       msg.sender,
       rewardSafeOwner
     );
+
+    bool isAllowedGnosisExecution = _equalBytes(data, encodedTransactionData) &&
+      signatures[keccak256(contractSignature)] &&
+      isAllowedGnosisRecipient(to);
+
+    bytes4 result = isAllowedGnosisExecution ||
+      (to == address(this)) ||
+      eip1271Contracts.contains(to)
```

Comment:
@alex-cardstack I missed something out here. Each OR condition should check `_equalBytes(data,encodedTransactionData)`. This is to ensure fidelity; so people don't lie about `to` param.  Think it would be as simple as just `require(_equalBytes(data,encodedTransactionData))`